stan/math/rev/functor/algebra_solver_powell.hpp

```diff
@@ -122,14 +122,17 @@ struct algebra_solver_vari : public vari {
  * @throw <code>std::domain_error</code> if the norm of the solution exceeds
  * the function tolerance.
  */
-template <typename F, typename T>
+template <typename F, typename T, require_eigen_vector_t<T>* = nullptr>
 Eigen::VectorXd algebra_solver_powell(
-    const F& f, const Eigen::Matrix<T, Eigen::Dynamic, 1>& x,
-    const Eigen::VectorXd& y, const std::vector<double>& dat,
-    const std::vector<int>& dat_int, std::ostream* msgs = nullptr,
-    double relative_tolerance = 1e-10, double function_tolerance = 1e-6,
+    const F& f, const T& x, const Eigen::VectorXd& y,
+    const std::vector<double>& dat, const std::vector<int>& dat_int,
+    std::ostream* msgs = nullptr, double relative_tolerance = 1e-10,
+    double function_tolerance = 1e-6,
     long int max_num_steps = 1e+3) {  // NOLINT(runtime/int)
-  algebra_solver_check(x, y, dat, dat_int, function_tolerance, max_num_steps);
+  const auto& x_eval = x.eval();
+  const auto& x_val = (value_of(x_eval)).eval();
```

Comment:
Why do we need to eval `value_of()`? Can the rest of this not take in a `holder` type?